### DISCUSSION TITLE
List all active clients

### DISCUSSION THREAD

**Discussion by skrystosik** [mongodb/mongo-ruby-driver#2838] [General]
Hi guys, 
I need to list and iterate through all existing clients from ruby code. Is there any hook we can use for that? I'm checking documentation and sourcecode but can't find it.

**Comment by comandeo** [comment]
Hi @skrystosik , the driver does not keep track of existing clients. However, you can use `ObjectSpace` module to get `Mongo::Client` objects:

```ruby
ObjectSpace.each_object(Mongo::Client) { |client| pp client }
```

**Reply by skrystosik** [reply]
Great, thank you!

**Reply by skrystosik** [reply]
I have one extra question, we're not able to hook subscriber before client is created. Can we check somehow if client has active sbuscriber to be sure that we do not start more than one at the time?

**Reply by comandeo-mongo** [reply]
If the "subscriber" means SDAM monitoring subscriber, you can use `sdam_proc` option to hook subscriber before client is created - see the example [here](https://www.mongodb.com/docs/ruby-driver/master/reference/monitoring/). Or you can use `Mongo::Monitoring::Global.subscribe`.

I do not think there is a "legal" option to check monitoring subscribers of a client.

**Reply by skrystosik** [reply]
I need to add COMMAND subscriber to client running in multiple instances (passenger phusion). In theory should work there but I see subscriber added to Mongo::Monitoring::Global (in our logs) but I don't see execution of subscriber. On our environment works fine :) I need to think about it little bit
